### PR TITLE
feat: create signup form for third step

### PR DESCRIPTION
**Pull request by fedasevich**
## Describe your changes

- Create SignUpThirdForm component.
- Add input validation.
- Add locales.

image
image
image

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://homecareaid.atlassian.net/browse/AC-6?atlOrigin=eyJpIjoiNWMwMmEwZTRlOTgwNGZlYTk1OTE5MWNiNDkxMWZhZmIiLCJwIjoiaiJ9l)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Comment by lyudad** [comment]
please update with second option https://mui.com/material-ui/react-text-field/

**Comment by fedasevich** [comment]
Change input component
Change formControl variant
Change select default value
image
image

File: src/components/sign-up-third/index.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import { useLocales } from 'src/locales';
+import { InferType } from 'yup';
+import { useSignUpThirdCountrySelectOptions } from './select-options';
+
+import { useSignUpThirdSchema } from './validation';
+import { AuthFormWrapper, ErrorMessage, NextButton, StyledForm } from './style';
+
+
+interface SignUpThirdFormProps {
+  onNext: unknown;
```
Comment (lyudad) [inline src/components/sign-up-third/index.tsx:17]: это не тип

File: src/components/sign-up-third/index.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    handleSubmit,
+    formState: { errors, isValid },
+  } = useForm<FormValues>({
+    resolver: yupResolver(signUpThirdSchema),
+    mode: 'onBlur',
+  });
+
+  const onSubmit = handleSubmit((data) => {
+    dispatch(saveAddressData(data))
+    alert(JSON.stringify(data))
```
Comment (lyudad) [inline src/components/sign-up-third/index.tsx:41]: не, алерты только на тестовых поректах и то редко, найти другой способ

File: src/components/sign-up-third/validation.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  const stateRequired: string = translate('signUpThirdForm.stateRequired');
+
+  const cityMaxLength: string = translate('signUpThirdForm.cityMaxLength');
+  const cityRequired: string = translate('signUpThirdForm.cityRequired');
+
+  const zipCodeMaxLength: string = translate('signUpThirdForm.zipCodeMaxLength');
+  const zipCodeRequired: string = translate('signUpThirdForm.zipCodeRequired');
+
+  const addressMaxLength: string = translate('signUpThirdForm.addressMaxLength');
+  const addressRequired: string = translate('signUpThirdForm.addressRequired');
```
Comment (lyudad) [inline src/components/sign-up-third/validation.ts:36]: зачем константы создавать? там же просто присваивание строки идет, только файл большой становится и имена полчаса придумывать нужно

File: src/pages/signup.tsx [hunk 4]
```diff
@@ -1,49 +1,49 @@
...
-
-import Step1Form from "src/components/sign-up/sign-up-first/SignUpForm1";
-import SignUpHeader from "src/components/reusable/header";
-
+import Step1Form from 'src/components/sign-up/sign-up-first/SignUpForm1';
+import SignUpThirdForm from 'src/components/sign-up-third';
+import SignUpHeader from 'src/components/reusable/header';
 import { useTranslation } from 'react-i18next';
 import { RootState } from 'src/redux/store';
 import { useSignUpMutation } from 'src/redux/api/userAPI';
```
Comment (lyudad) [inline src/pages/signup.tsx:8]: https://www.notion.so/Git-rules-849f8ca8b3db4ded926d762c6fff8798 1.1.14 не времательно сделано

File: src/pages/signup.tsx [hunk 5]
```diff
@@ -1,49 +1,49 @@
...
 
+const SignUp = (): JSX.Element => {
+  const [step, setStep] = useState(1);
```
Comment (lyudad) [inline src/pages/signup.tsx:11]: типы

File: src/pages/signup.tsx [hunk 6]
```diff
@@ -1,49 +1,49 @@
...
-        .then(() => {
-          setStep(prevStep => prevStep + 1);
-        })
-        .catch((error) => {
-          throw new Error(error)
-        });
-    };
-    
-    const { t } = useTranslation();
+  const { t } = useTranslation();
```
Comment (lyudad) [inline src/pages/signup.tsx:37]: почему хук после хендлеров? должна быть структура в каждом компоненте одинаковая

File: src/components/sign-up-third/index.tsx (hunk 1 above)
Comment (fedasevich) [reply src/components/sign-up-third/index.tsx]: done

File: src/components/sign-up-third/index.tsx (hunk 2 above)
Comment (fedasevich) [reply src/components/sign-up-third/index.tsx]: done

File: src/pages/signup.tsx (hunk 6 above)
Comment (ProggerOleg) [reply src/pages/signup.tsx]: done

File: src/pages/signup.tsx (hunk 5 above)
Comment (ProggerOleg) [reply src/pages/signup.tsx]: done

File: src/pages/signup.tsx (hunk 4 above)
Comment (ProggerOleg) [reply src/pages/signup.tsx:9]: done

File: src/components/sign-up-third/validation.ts (hunk 3 above)
Comment (ProggerOleg) [reply src/components/sign-up-third/validation.ts]: done

File: src/pages/signup.tsx [hunk 7]
```diff
@@ -1,49 +1,51 @@
...
 import { useSignUpMutation } from 'src/redux/api/userAPI';
 
+import Step1Form from 'src/components/sign-up/sign-up-first/SignUpForm1';
+import SignUpThirdForm from 'src/components/sign-up-third';
+import SignUpHeader from 'src/components/reusable/header';
+
+const SignUp = (): JSX.Element => {
+  const { t } = useTranslation();
+
+  const [step, setStep] = useState<number>(1);
```
Comment (lyudad) [inline src/pages/signup.tsx:15]: https://www.notion.so/Git-rules-849f8ca8b3db4ded926d762c6fff8798 1.1.4

**Comment by lyudad** [comment]
no tests

File: src/pages/signup.tsx (hunk 7 above)
Comment (fedasevich) [reply src/pages/signup.tsx]: done

File: src/components/sign-up-third/SignUpThirdForm.tsx [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+export default function SignUpThirdForm({ onNext }: SignUpThirdFormProps): JSX.Element {
+  const { translate } = useLocales();
+  const dispatch = useAppDispatch();
+  const signUpThirdSchema = useSignUpThirdSchema();
+  const { countries } = useSignUpThirdCountrySelectOptions();
+  const initialAddressValues = useTypedSelector((state) => state.address.addressData);
+
+  type FormValues = InferType<typeof signUpThirdSchema>;
```
Comment (lyudad) [inline src/components/sign-up-third/SignUpThirdForm.tsx:26]: why type inside the component?

File: src/components/sign-up-third/SignUpThirdForm.tsx [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+    handleSubmit,
+    watch,
+    formState: { errors, isValid },
+  } = useForm<FormValues>({
+    resolver: yupResolver(signUpThirdSchema),
+    mode: 'onBlur',
+    defaultValues: initialAddressValues,
+  });
+
+  const selectedCountry = watch('country');
```
Comment (lyudad) [inline src/components/sign-up-third/SignUpThirdForm.tsx:39]: send me where you found such an implementation example. I don't want to see any constant in the component. it lead to reinitialization with each rerender

File: src/components/sign-up-third/SignUpThirdForm.tsx (hunk 8 above)
Comment (fedasevich) [reply src/components/sign-up-third/SignUpThirdForm.tsx]: fixed

File: src/components/sign-up-third/SignUpThirdForm.tsx (hunk 9 above)
Comment (fedasevich) [reply src/components/sign-up-third/SignUpThirdForm.tsx]: fixed

**Review by lyudad — APPROVED** [review]

**Comment by lyudad** [comment]
Screenshot 2023-11-21 at 20 50 14

**Pull request merged**